definitions/src/sql_data_type.rs

```diff
@@ -0,0 +1,49 @@
+use num_derive::FromPrimitive;
+
+#[allow(non_camel_case_types)]
+#[repr(i16)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy, FromPrimitive)]
+pub enum SqlDataType {
+    SQL_UNKNOWN_TYPE = 0,
+    SQL_CHAR = 1,
+    SQL_NUMERIC = 2,
+    SQL_DECIMAL = 3,
+    SQL_INTEGER = 4,
+    SQL_SMALLINT = 5,
+    SQL_FLOAT = 6,
+    SQL_REAL = 7,
+    SQL_DOUBLE = 8,
+    SQL_DATETIME = 9,
+    SQL_VARCHAR = 12,
+    SQL_TYPE_DATE = 91,
+    SQL_TYPE_TIME = 92,
+    SQL_TYPE_TIMESTAMP = 93,
+    SQL_TIME_OR_INTERVAL = 10,
```

Comment:
Looking at the headers, the values differs depending on the odbc version.
Since we are an ODBC 3.0+ driver, we should be able to safely say that this is SQL_INTERVAL or if you think that it best to highlight that the value is used under 2 different "aliases" then just a comment on top would do.

Reply:
@terakilobyte and I had talked about this a little bit -- I think keeping odbc 3 conventions makes sense, and so SQL_INTERVAL here makes sense to me. A followup ticket that more generally considers odbc version / feature flags / etc now that we are no longer tied to odbc-sys likely makes sense, too

Reply:
Works for me 👍